python_modules/dagster-graphql/dagster_graphql/schema/backfill.py

```diff
@@ -524,6 +531,72 @@ def resolve_hasResumePermission(self, graphene_info: ResolveInfo) -> bool:
     def resolve_user(self, _graphene_info: ResolveInfo) -> Optional[str]:
         return self._backfill_job.user
 
+    def resolve_logEvents(self, graphene_info: ResolveInfo, cursor: Optional[str] = None):
+        # very unsure how the cursor gets passed through, especially if called from get_backfills which also has a cursor
+        from ..schema.instigation import (
+            GrapheneInstigationEvent,
+            GrapheneInstigationEventConnection,
+        )
+        from ..schema.logs.log_level import GrapheneLogLevel
+
+        backfill_id = self._backfill_job.backfill_id
+        # TODO find a better way to keep this in sync. maybe store as part of the asset backfill data?
+        backfill_log_key_prefix = ["backfill", backfill_id]
+
+        instance = graphene_info.context.instance
+
+        if not isinstance(instance.compute_log_manager, CapturedLogManager):
+            return GrapheneInstigationEventConnection(events=[], cursor="", hasMore=False)
+
+        # TODO - need to gate to plus only
+
+        log_keys = sorted(
+            instance.compute_log_manager.get_log_keys_for_log_key_prefix(
+                backfill_log_key_prefix, io_type=ComputeIOType.STDERR
+            )
+        )
+        log_key_to_fetch = cursor.split("/") if cursor else log_keys[0]
+        log_data = instance.compute_log_manager.get_log_data(log_key_to_fetch)
```

Comment:
right now pagination is done at the log file level. you basically get one "tick" of logs at a time. i'm planning to update this so that the logs act more like a continuous stream from a single file where you request N bytes and we automatically progress through the files N bytes at a time

Reply:
server determined page/chunk size isn't crazy to stick with - don't expect the front-end to pick a better size than the server can